### DISCUSSION TITLE
The dialog box prompting to save data when closing Visual Studio Code

### DISCUSSION THREAD

**Discussion by feeelin** [microsoft/vscode-discussions#2407] [Extension Development QnA] [answered]
In our VS Code extension, we need to ask the user whether they want to save data when closing Visual Studio Code. However, we couldn't find this functionality in the API. It's possible to delay closing for up to 10 seconds to terminate background tasks, but fully suspending the editor's closure for user confirmation doesn't seem possible.

Is there an existing way to implement such a dialog? If not, are there plans to introduce this capability in the future?

**Comment by isidorn** [comment, marked as answer]
The extension APIs do not provide hooks for intercepting or delaying the application shutdown process.
I do not think there are plans right now. But if that change I can update this issue.
